airflow/models.py

```diff
@@ -1262,7 +1262,7 @@ def get_dagrun(self, session):
         return dr
 
     @provide_session
-    def run(
+    def set_running_state_before_execution(
```

Comment:
I don't think the name is great, but try_setting_task_to_running may give misleading impressions, so I'll update the documentation and let that do the talking rather than give readers a sense of false confidence.